tests/src/Simple/Threading/Threading.cs

```diff
@@ -758,6 +789,282 @@ public static void MutexMaximumReacquireCountTest()
+        Assert.True(testJobCompleted.WaitOne(ThreadTestHelpers.ExpectedMeasurableTimeoutMilliseconds));
+    }
+
+    [Fact]
+    public static void ThreadLocalWorkQueueDepletionTest()
+    {
+        ManualResetEvent e0 = new ManualResetEvent(false);
+        int count = 0;
+        int maxCount = Environment.ProcessorCount * 64;
+        object syncRoot = new object();
+        void Job()
+        {
+            if(Interlocked.Increment(ref count) >= maxCount)
+            {
+                e0.Set();
+            }
+            else
+            {
+                Task.Factory.StartNew(Job);
+                Task.Factory.StartNew(Job);
+            }
+        }
+        Task.Factory.StartNew(Job);
+        e0.WaitOne();
+    }
+
+    [Fact]
+    public static void WorkerThreadStateReset()
+    {
+        var cultureInfo = new CultureInfo("pt-BR");
+        var expectedCultureInfo = CultureInfo.CurrentCulture;
+        var expectedUICultureInfo = CultureInfo.CurrentUICulture;
+        int count = 0;
+        AutoResetEvent e0 = new AutoResetEvent(false);
+        for(int i = 0; i < Environment.ProcessorCount; ++i)
+        {
+            Task.Factory.StartNew(() => {
+                CultureInfo.CurrentCulture = cultureInfo;
+                CultureInfo.CurrentUICulture = cultureInfo;
+                Thread.CurrentThread.Priority = ThreadPriority.AboveNormal;
+                if(Interlocked.Increment(ref count) == Environment.ProcessorCount)
+                {
+                    e0.Set();
+                }
+            });
+        }
+        Assert.True(e0.WaitOne(ThreadTestHelpers.UnexpectedTimeoutMilliseconds));
+        // Run the test again to make sure we can reuse the threads.
+        count = 0;
+        for(int i = 0; i < Environment.ProcessorCount; ++i)
+        {
+            Assert.Equal(expectedCultureInfo, CultureInfo.CurrentCulture);
+            Assert.Equal(expectedUICultureInfo, CultureInfo.CurrentUICulture);
+            Assert.Equal(ThreadPriority.Normal, Thread.CurrentThread.Priority);
+            Task.Factory.StartNew(() => {
+                if(Interlocked.Increment(ref count) == Environment.ProcessorCount)
+                {
+                    e0.Set();
+                }
+            });
+        }
+        Assert.True(e0.WaitOne(ThreadTestHelpers.UnexpectedTimeoutMilliseconds));
+    }
+
+    [Fact]
+    public static void GlobalWorkQueueDepletionTest()
+    {
+        ManualResetEvent e0 = new ManualResetEvent(false);
+        int count = 0;
+        int maxCount = Environment.ProcessorCount * 64;
+        object syncRoot = new object();
+        void Job(object _)
+        {
+            if(Interlocked.Increment(ref count) >= maxCount)
+            {
+                e0.Set();
+            }
+            else
+            {
+                ThreadPool.QueueUserWorkItem(Job);
+                ThreadPool.QueueUserWorkItem(Job);
+            }
+        }
+        ThreadPool.QueueUserWorkItem(Job);
```

Comment:
This should be combined with the task one, to mix local and global queues together in one depletion test. For QueueUserWorkItem, it would be more direct to just have a loop to queue them all.